Userland/Utilities/json.cpp

```diff
@@ -121,33 +149,181 @@ void print(JsonValue const& value, int spaces_per_indent, int indent, bool use_c
         out("\033[0m");
 }
 
-JsonValue query(JsonValue const& value, Vector<StringView>& key_parts, size_t key_index)
-{
-    if (key_index == key_parts.size())
-        return value;
-    auto key = key_parts[key_index++];
+class WildcardKey : public MultiValueKey {
+public:
+    static NonnullRefPtr<WildcardKey> parse(StringView const&)
+    {
+        return make_ref_counted<WildcardKey>();
+    }
 
-    if (key == "*"sv) {
+    Vector<JsonValue> select(JsonValue const& value) override
+    {
         Vector<JsonValue> matches;
         if (value.is_object()) {
-            value.as_object().for_each_member([&](auto&, auto& member_value) {
-                matches.append(query(member_value, key_parts, key_index));
+            value.as_object().for_each_member([&](auto const&, auto& member_value) {
+                matches.append(member_value);
             });
         } else if (value.is_array()) {
-            value.as_array().for_each([&](auto& member) {
-                matches.append(query(member, key_parts, key_index));
-            });
+            matches = value.as_array().values();
+        }
+        return matches;
+    }
+};
+
+class SliceKey : public MultiValueKey {
+private:
+    int m_start;
+    int m_end;
+    int m_step;
+
+public:
+    SliceKey(int start, int end, int step)
+        : m_start(start)
+        , m_end(end)
+        , m_step(step)
+    {
+    }
+
+    static ErrorOr<NonnullRefPtr<SliceKey>> parse(StringView const& key)
+    {
+        auto contents = key.substring_view(1, key.length() - 2);
+        auto components = contents.split_view(':', SplitBehavior::KeepEmpty);
+
+        int start = INT_MIN;
+        int end = INT_MAX;
```

Comment:
Instead of `X_MAX` or `X_MIN`, prefer to use `AK::NumericLimits

Reply:
Oh good tip!